jep/0000/README.adoc

```diff
@@ -0,0 +1,366 @@
+
+Having a centralized release system made available to plugin maintainers also provides additional 
+confidence that security best practices are being followed. 
+
+Continuous delivery from trusted CI is something which plugin maintainers can opt in to, but is 
+not required. If a plugin maintainer chooses to continue to follow their own path for releasing 
+versions of their code, they remain free to do so.
+
+== Reasoning
+
+=== What's with the version numbering?
+
+Jenkins plugin maintainers are already familiar with the way that Incrementals appends a commit 
+number, plus a SHA, to version numbers. These mechanically-generated version numbers offer 
+the ability to predictably sort them, so that external systems, such as the Jenkins update center, 
+can correctly publish the "newest" version. The addition of a merge SHA also allows for at-a-glance 
+feedback to tell people which commits went into the release.
+
+WARNING: We know the problem of consistent version numbering, and the resulting ability to 
+sort versions, is important. The scheme discussed here is only one of several possible 
+choices. We expect some _lively debate_ around this topic. But it's very important to ensure 
+that versions can be easily sorted by systems such as the Jenkins update center. Also under 
+consideration is defining a fixed number of fields in version numbers, and considering any 
+deviation from that a validation failure.
+
+=== Why YAML?
+YAML is becoming increasingly common in the Jenkins community, for many reasons. YAML is:
+
+* Human readable
+* In use by things like the Kubernetes plugin
+* Easily parsed by any number of publicly available libraries
+
+=== Webhooks and the Webhook Receiver
+Webhooks will be used to trigger the builds on the trusted CI server. The hooks themselves 
+will be configured on a per-repository basis. The reasons for choosing individual hooks, 
+as opposed to an organization wide hook, are:
+
+* Security of the build systems. Because this trusted CI server will reside on a non-public 
+network, there needs to be a proxy server of sorts, which can accept these webhooks and send 
+them, securely, to the trusted CI server.
+
+
+* Traceability. An organization-wide webhook would send those hooks to the receiver far more 
+often than is actually necessary. In the event that something goes wrong with this process - 
+the receiver goes down, there is a network outage, etc. - it will be easier for the Infra 
+team to triage problems if they're only looking at hooks from plugins which are intended 
+to be released via CD.
+
+* Flexibility for plugin maintainers. Let's say that Karl maintains two plugins - his archaic 
+`old-timey-plugin`, and his fancy new `karl-plugin` which we've already talked about. If 
+Karl wants to use CD on karl-plugin, but continue to release `old-timey-plugin` from his laptop 
+like he's always done, he can do so, by only setting the webhook up on `karl-plugin`.
```

Comment:
Does not make sense to me. You are already proposing to gate autorelease by an explicit flag.

Reply:
On revisiting this you're right. Was unnecessary and has been deleted.